idrstream/merge_CP_DP.py

```diff
@@ -0,0 +1,162 @@
+"""
+This file holds the functions necessary for merging IDR_stream CP and DP outputs.
+As the IDR_stream output follows the pycytominer output format, these merge functions should work for any pycytominer output data.
+"""
+
+import pandas as pd
+import pathlib
+import math
+
+
+def merge_CP_DP_batch_data(
+    cp_batch_data: pd.DataFrame, dp_batch_data: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    merge dataframes for IDR_stream output with CP and DP features
+    the two features dataframes should have aligned location metadata (plate, well, frame, etc) and the same number of rows (cells)
+
+    Parameters
+    ----------
+    cp_batch_data : pd.DataFrame
+        idrstream_cp batch output
+    dp_batch_data : pd.DataFrame
+        idrstream_dp batch output
+
+    Returns
+    -------
+    pd.DataFrame
+        merged batch data with metadata, CP features, and DP features
+
+    Raises
+    ------
+    IndexError
+        cp and dp dataframes have different number of rows (cells)
+    """
+
+    # check batch data have same number of rows (cells)
+    # if batch data have different number of cells, raise an error because they must not have close segmentations
+    if cp_batch_data.shape[0] != dp_batch_data.shape[0]:
+        raise IndexError("Batch data have different number of rows (cells)!")
+
+    # hide warning for pandas chained assignment
+    pd.options.mode.chained_assignment = None
```

Comment:
Seeing the work to remove warnings or raises, consider providing details about why this is necessary. If this isn't necessary for the work and you could use other methods instead, I recommend doing this to avoid the custom controls of the warnings.

Reply:
Done in https://github.com/WayScience/IDR_stream/pull/15/commits/e6a24c15f9d0abe772e43469f902a40d7431a0c2